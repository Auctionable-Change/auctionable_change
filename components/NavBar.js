import React from 'react'
import { View, Image } from "react-native";
import { StyleSheet } from "react-native";
import { TouchableWithoutFeedback } from "react-native-gesture-handler";



const NavBar = ({ navigation }) => {

  return (
    <View style={styles.navBar}>
      <TouchableWithoutFeedback
        onPress={() => navigation.navigate("Current Listings")}
      >
        <Image
          accessibilityRole="button"
          id="search"
          source={require("../assets/icons/search.png")}
          style={styles.icons}
        />
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        onPress={() => navigation.navigate("Post Item")}
      >
        <Image
          accessibilityRole="button"
          id="add"
          source={require("../assets/icons/add.png")}
          style={styles.icons}
        />
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback>
        <Image
          accessibilityRole="button"
          id="profile"
          source={require("../assets/icons/profile.png")}
          style={styles.icons}
        />
      </TouchableWithoutFeedback>
    </View>
  );
}

const styles = StyleSheet.create({
  navBar: {
    width: "100%",
    backgroundColor: "white",
    position: "absolute",
    bottom: "0%",
    height: "11%",
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-around",
    padding: 10,
  },
  icons: {
    width: 45,
    height: 45,
  },
});

export default NavBar;
