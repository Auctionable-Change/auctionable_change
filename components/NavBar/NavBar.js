import React from "react";
import { View, Image } from "react-native";
import { StyleSheet } from "react-native";
import { TouchableWithoutFeedback } from "react-native-gesture-handler";

const NavBar = ({ navigation }) => {
  return (
    <View style={styles.navBar}>
      <TouchableWithoutFeedback onPress={() => navigation.navigate("Listings")}>
        <Image
          accessibilityRole="button"
          id="search"
          source={require("../../assets/icons/search.png")}
          style={styles.icons}
        />
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        onPress={() => navigation.navigate("Post Item")}
      >
        <Image
          accessibilityRole="button"
          id="add"
          source={require("../../assets/icons/add.png")}
          style={styles.icons}
        />
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback onPress={() => navigation.navigate("Home")}>
        <Image
          accessibilityRole="button"
          id="home"
          source={require("../../assets/icons/home.png")}
          style={styles.icons}
        />
      </TouchableWithoutFeedback>
    </View>
  );
};

const styles = StyleSheet.create({
  navBar: {
    width: "100%",
    backgroundColor: "#2cb833",
    position: "absolute",
    bottom: "0%",
    height: "8%",
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-around",
    padding: 8,
    borderTopColor: "#dddddd",
    borderTopWidth: 1,
  },
  icons: {
    width: 30,
    height: 30,
  },
});

export default NavBar;
