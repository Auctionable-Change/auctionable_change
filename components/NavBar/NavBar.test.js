import NavBar from "./NavBar";
import React from "react";
import renderer from "react-test-renderer";


describe("NavBar", () => {
  it("should render three icons component", () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree.children.length).toBe(3);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render a search icon first", () => {
    const tree = renderer.create(<NavBar />).toJSON();
    const searchIconId = tree.children[0].children[0].children[0].props.id;
    expect(searchIconId).toBe("search");
  });
  it("should render a plus icon second", () => {
    const tree = renderer.create(<NavBar />).toJSON();
    const addIconId = tree.children[1].children[0].children[0].props.id;
    expect(addIconId).toBe("add");
  });
  it("should render a profile icon third", () => {
    const tree = renderer.create(<NavBar />).toJSON();
    const addIconId = tree.children[2].children[0].children[0].props.id;
    expect(addIconId).toBe("home");
  });

});
