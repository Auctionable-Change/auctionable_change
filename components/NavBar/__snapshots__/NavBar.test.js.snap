// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderTopColor": "#dddddd",
      "borderTopWidth": 1,
      "bottom": "0%",
      "display": "flex",
      "flexDirection": "row",
      "height": "8%",
      "justifyContent": "space-around",
      "padding": 8,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
    >
      <Image
        accessibilityRole="button"
        id="search"
        source={1}
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
    >
      <Image
        accessibilityRole="button"
        id="add"
        source={1}
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
    >
      <Image
        accessibilityRole="button"
        id="home"
        source={1}
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
  </RNGestureHandlerButton>
</View>
`;
